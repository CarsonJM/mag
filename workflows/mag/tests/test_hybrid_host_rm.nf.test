nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_hybrid_host_rm"

    test("Parameters: long reads & host removal included") {

        when {
            params {
                input                       = 'https://raw.githubusercontent.com/nf-core/test-datasets/mag/samplesheets/samplesheet.hybrid_host_rm.csv'
                host_fasta                  = "https://raw.githubusercontent.com/nf-core/test-datasets/mag/host_reference/genome.hg38.chr21_10000bp_region.fa"
                bowtie2_fix_cpu_1           = true
                keep_phix                   = true
                skip_clipping               = true
                centrifuge_db               = null
                kraken2_db                  = null
                skip_krona                  = true
                skip_megahit                = true
                skip_spades                 = true
                skip_spadeshybrid           = false
                spadeshybrid_fix_cpus       = 2
                skip_quast                  = true
                skip_prodigal               = true
                skip_binning                = true
                skip_binqc                  = true
                skip_gtdbtk                 = true
                skip_prokka                 = true
                skip_metaeuk                = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out,
                ).match()}
            )
        }
    }
}
