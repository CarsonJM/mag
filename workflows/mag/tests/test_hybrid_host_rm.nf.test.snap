{
    "Parameters: long reads & host removal included": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "minigut_run0_host_removed.unmapped_1.fastq.gz:md5,48ab5cdbf8567cf6023bd304fcf896a9",
                            "minigut_run0_host_removed.unmapped_2.fastq.gz:md5,b4246225aa4a15496991ac7c6b58563f"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdesHybrid"
                        },
                        "SPAdesHybrid-minigut_scaffolds.fasta:md5,d3c5e27f011a08daea597f9d2650436d"
                    ]
                ],
                "10": [
                    "versions.yml:md5,1bd6bbec156534880e04e0320dbce063",
                    "versions.yml:md5,59050473c9f749b269ad7079fd4693df",
                    "versions.yml:md5,5de8983775d3ef64cdbec6f382852619",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,b94aaba1559ac9deaf7339181c398405",
                    "versions.yml:md5,cc143ba1b806a335a9963a434c887eba",
                    "versions.yml:md5,d13508b80e270ac6f1543725b431a682",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0"
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "assemblies": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdesHybrid"
                        },
                        "SPAdesHybrid-minigut_scaffolds.fasta:md5,d3c5e27f011a08daea597f9d2650436d"
                    ]
                ],
                "bin_summary": [
                    
                ],
                "cat": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    
                ],
                "refined_unbins": [
                    
                ],
                "short_reads": [
                    [
                        {
                            "id": "minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "minigut_run0_host_removed.unmapped_1.fastq.gz:md5,48ab5cdbf8567cf6023bd304fcf896a9",
                            "minigut_run0_host_removed.unmapped_2.fastq.gz:md5,b4246225aa4a15496991ac7c6b58563f"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1bd6bbec156534880e04e0320dbce063",
                    "versions.yml:md5,59050473c9f749b269ad7079fd4693df",
                    "versions.yml:md5,5de8983775d3ef64cdbec6f382852619",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,b94aaba1559ac9deaf7339181c398405",
                    "versions.yml:md5,cc143ba1b806a335a9963a434c887eba",
                    "versions.yml:md5,d13508b80e270ac6f1543725b431a682",
                    "versions.yml:md5,d654471556992ae3c82a7fa792c90dc0"
                ]
            }
        ],
        "timestamp": "2024-01-17T10:03:17.006107632"
    }
}